presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetRecordCursorProvider.java

```diff
@@ -83,7 +80,10 @@ public ParquetRecordCursorProvider(boolean useParquetColumnNames)
         }
 
         // are all columns supported by Parquet code
-        List<HiveColumnHandle> unsupportedColumns = ImmutableList.copyOf(filter(columns, not(isParquetSupportedType())));
+        List<HiveColumnHandle> unsupportedColumns = columns.stream()
+                .filter(columnHandle -> !columnHandle.isPartitionKey())
+                .filter(columnHandle -> columnHandle.getHiveType().equals(HiveType.HIVE_DATE))
+                .collect(toList());
```

Comment:
@cberner  I was previously told to just use `toList` instead in presto-hive. And if you search for usage of `toList` in presto-hive, you can find a lot of them in HiveMetadata, HivePageSink, HiveType, and HiveWriteUtils.